Automattic/newspack-plugin
# feat: start new RAS settings page

### All Submissions:






### Changes proposed in this Pull Request:



Scaffolds a new page layout for the RAS settings page according to the latest i2 designs. The list of prerequisites and all copy/links are TBD, so this PR contains a list of a few known requirements and a boilerplate config object to serve as a template to add more.

Because discussions are ongoing about what to do about the existing RAS settings on this page, these are simply hidden behind an "advanced settings" flag to show/hide those settings. This advanced settings section will need to be redesigned in a future PR.

The `ActionCard` component to kick off the Campaigns default prompts/segments setup is a placeholder, and is non-interactive for now.

### How to test the changes in this Pull Request:

1. Visit the **Newspack > Engagement > Reader Activation** admin page and ensure that the prerequisites are shown according to the config array in `includes/reader-activation/class-reader-activation.php` and that all the links and buttons work. Feel free to copy the existing config objects and create your own to test different scenarios. This is the state of the page before all RAS prerequisites are completed:

<img width="1195" alt="Screen Shot 2023-03-29 at 3 10 49 PM" src="https://user-images.githubusercontent.com/2230142/228668398-c673668c-3544-4895-ba31-52a8188c269a.png">

2. Open the Advanced Settings section and ensure that all of these settings options still work as in `master`, for now. Note that some fields have been removed based on discussions that they should always be set one way or another if RAS is enabled. This section will need to be redesigned in a future PR.

### Other information:
